### PR TITLE
Use MCOIMAPCheckAccountOperation as return type for ObjC IMAP Session::checkAccount operation

### PR DESCRIPTION
**Pull request by disaykin**

### CODE REVIEW

**Comment by dinhviethoa** [comment]
I'm not sure why you need to return an explicit type?

**Comment by disaykin** [comment]
I need information about loginResponse. This information returns only via userInfo field now. And only in AccountCheckOperation callback.

Without my changes i don't receive this field in error's userInfo.

But now I think I'm tested in the previous build of the library (without your latest additions of AccountCheckOperation functionality).

So I revert one commit.

File: src/objc/imap/MCOIMAPCheckAccountOperation.mm [hunk 1]
```diff
@@ -62,8 +62,9 @@ - (void) operationCompleted
     } else {
         NSError * error = [NSError mco_errorWithErrorCode:op->error()];
         if (op->loginResponse() != NULL) {
-            NSDictionary * userInfo = @{@"IMAPServerError": MCO_TO_OBJC(op->loginResponse())};
-            error = [NSError errorWithDomain:[error domain] code:[error code] userInfo:userInfo];
+            NSMutableDictionary * userInfo = [[error userInfo] mutableCopy];
+            userInfo[@"IMAPServerError"] = MCO_TO_OBJC(op->loginResponse());
+            error = [NSError errorWithDomain:[error domain] code:[error code] userInfo:[userInfo copy]];
```
Comment (dinhviethoa) [inline src/objc/imap/MCOIMAPCheckAccountOperation.mm]: use `error = [NSError errorWithDomain:[error domain] code:[error code] userInfo:userInfo];` instead.

**Comment by dinhviethoa** [comment]
one minor change and then, it's ready to merge.

**Comment by disaykin** [comment]
Done

**Pull request merged**